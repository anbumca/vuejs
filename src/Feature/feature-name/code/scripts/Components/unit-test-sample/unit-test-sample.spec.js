import { shallowMount } from '@vue/test-utils'
import Hello from './unit-test-sample.vue'

test('Hello', () => {
  // render the component
  const wrapper = shallowMount(Hello)

  // should not allow for `username` less than 7 characters, excludes whitespace
  wrapper.setData({ username: 'A'.repeat(7) })

  // assert the error is rendered
  expect(wrapper.find('.error').exists()).toBe(true)

  // update the name to be long enough
  wrapper.setData({ username: 'Anbarasan' })

  // assert the error has gone away
  expect(wrapper.find('.error').exists()).toBe(true)
  expect(wrapper).toMatchSnapshot();
})