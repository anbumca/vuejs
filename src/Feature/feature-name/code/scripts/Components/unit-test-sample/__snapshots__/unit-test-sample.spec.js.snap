// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello 1`] = `
<div><input>
  <div class="error">
    Please enter a longer username
  </div>
</div>
`;
